docs/getting_started.rst

```diff
@@ -54,6 +54,94 @@ Connecting with a client
     # Disconnect from the server
     client.disconnect()
 
+**************
+Authentication
+**************
+
+When constructing a ``Cloudant`` client, you can authenticate using the
+[cookie authentication](http://guide.couchdb.org/editions/1/en/security.html#cookies) functionality.
+The server will always attempt to automatically renew the cookie
+shortly before its expiry. However, if the client does not send a
+request to the server during this renewal window and
+``auto_renew=False`` then the cookie is not renewed.
+
+Using ``auto_renew=True`` will attempt to renew the cookie at
+any point during the lifetime of the session when either of the
+following statements hold true:
+
+- The server returns a ``credentials_expired`` error message.
+- The server returns a ``401 Unauthorized`` status code.
+- The server returns a ``403 Forbidden`` status code.
+
+.. code-block:: python
+
+    # Create client using auto_renew to automatically renew expired cookie auth
+    client = Cloudant(USERNAME, PASSWORD, url='https://acct.cloudant.com',
+                     connect=True,
+                     auto_renew=True)
+
+****************
+Resource sharing
+****************
+
+The ``Cloudant`` or ``CouchDB`` client objects make HTTP calls using the ``requests`` library.
+``requests`` uses the [urllib3](https://pypi.python.org/pypi/urllib3) library which features
+connection pooling and thread safety.
+
+Connection pools can be managed by using the ``requests`` library's
+[HTTPAdapter](https://github.com/kennethreitz/requests/blob/master/requests/adapters.py#L78)
+when constructing a ``Cloudant`` or ``ClouchDB`` client instance.
+The default number set by the ``urllib3`` library for cached connection pools is 10.
+Use the ``HTTPAdapter`` argument ``pool_connections`` to set the number of
+urllib3 connection pools to cache, and the ``pool_maxsize`` argument to set the
+maximum number of connections to save in the pool.
+
+Although the ``client`` session is documented as thread safe and it's possible for a
+static ``client`` to be accessible by multiple threads, there are still cases that do not
+guarantee thread safe execution.  It's recommended to use one ``client`` object per thread.
+
+.. code-block:: python
+
+    # Create client with 15 cached pool connections and a max pool size of 100
+    httpAdapter = HTTPAdapter(pool_connections=15, pool_maxsize=100)
+    client = Cloudant(USERNAME, PASSWORD, url='https://acct.cloudant.com'
+                     connect=True,
+                     adapter=httpAdapter)
+
+Note: Idle connections within the pool may be terminated by the server, so will not remain open
+indefinitely meaning that this will not completely remove the overhead of creating new connections.
+
+Using library in app server environment
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This library can be used in an app server, and the example
+below shows how to use ``client`` in a ``flask`` app server.
+
+.. code-block:: python
```

Comment:
I'm not familiar with flask, but my reading of this is that each defined `@app.route` is a HTTP handler and they can be serviced on multiple threads. Due to the ambiguity of requests/urllib3 around thread-safe cookie handling we are saying it is unsafe to access a single client object from multiple threads and hence we can't just store a cloudant `client` for use by all the http handlers. Instantiating a `client` on each handler call obviously has a performance penalty.

The alternative solution in my mind is to create a pool of `client` resources and lease one in the http handler? Similar to the way a  JDBC connection pool operates, does flask offer that kind of resource pooling? If it was done that way then the `client`s would be instantiated in the resource pool and used by only 1 HTTP handler thread at a time?